src/GrasshopperTeklaDrawingLink/Components/TeklaComponentBase.cs

```diff
@@ -100,5 +102,10 @@ protected void SetLastParameterAsOptional(GH_InputParamManager pManager, bool op
             if (optional)
                 pManager[pManager.ParamCount - 1].Optional = true;
         }
+        protected void SetParametersAsOptional(GH_InputParamManager pManager, List<int> indeces) 
+        {
+            for(int i = 0; i<indeces.Count; i++)
+                pManager[indeces[i]].Optional=true;
+        }
```

Comment:
Very good idea for utilization of returned indicies from GH AddParameter. Clever one :)

But please, fix the typo (indeces -> indicies) and transform this _for_ loop into _foreach_. It will increase readability of this method.
Now we accept input named indeces, then we are looping over every elementy in this list, to take its content and use it as index in another object - looks too complicated. _Foreach_ will help.
Please change also List<int> into IEnumerable<int>. When you will switch into foreach then generic collection will be enough.